forage-android/src/main/java/com/joinforage/forage/android/vault/ForagePinSubmitter.kt

```diff
@@ -0,0 +1,121 @@
+package com.joinforage.forage.android.vault
+
+import android.content.Context
+import com.joinforage.forage.android.VaultType
+import com.joinforage.forage.android.core.StopgapGlobalState
+import com.joinforage.forage.android.core.telemetry.Log
+import com.joinforage.forage.android.model.EncryptionKeys
+import com.joinforage.forage.android.model.PaymentMethod
+import com.joinforage.forage.android.network.ForageConstants
+import com.joinforage.forage.android.network.NetworkService
+import com.joinforage.forage.android.network.OkHttpClientBuilder
+import com.joinforage.forage.android.network.model.ForageApiResponse
+import com.joinforage.forage.android.network.model.UnknownErrorApiResponse
+import com.joinforage.forage.android.ui.ForagePINEditText
+import okhttp3.HttpUrl
+import okhttp3.HttpUrl.Companion.toHttpUrlOrNull
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.Request
+import okhttp3.RequestBody
+import okhttp3.RequestBody.Companion.toRequestBody
+import org.json.JSONObject
+
+internal class ForagePinSubmitter(
+    context: Context,
+    foragePinEditText: ForagePINEditText,
+    logger: Log
+) : AbstractVaultSubmitter<ForageApiResponse<String>>(
+    context = context,
+    foragePinEditText = foragePinEditText,
+    logger = logger,
+    vaultType = VaultType.FORAGE_VAULT_TYPE
+) {
+    // x-key header is not applicable to rosetta
+    override fun parseEncryptionKey(encryptionKeys: EncryptionKeys): String {
+        return ""
+    }
+
+    override suspend fun submitProxyRequest(vaultProxyRequest: VaultProxyRequest): ForageApiResponse<String> {
+        return try {
+            val apiUrl = buildVaultUrl(vaultProxyRequest.path)
+            val baseRequestBody = buildRequestBody(vaultProxyRequest)
+            val requestBody = buildForageVaultRequestBody(foragePinEditText, baseRequestBody)
+
+            val request = Request.Builder()
+                .url(apiUrl)
+                .post(requestBody)
+                .build()
+
+            val rosettaHeaderValues = vaultProxyRequest.params!!
+
+            val okHttpClient = OkHttpClientBuilder.provideOkHttpClient(
```

Comment:
`provideOkHttpClient` ensures idempotency key, trace id and merchant ID are provided as headers to the HTTP request.